contrib/tap_sinks/udp_sink/source/udp_sink_impl.h

```diff
@@ -0,0 +1,66 @@
+#pragma once
+
+#include "envoy/config/tap/v3/common.pb.h"
+
+#include "source/common/network/socket_impl.h"
+#include "source/common/network/socket_interface.h"
+#include "source/common/network/udp_packet_writer_handler_impl.h"
+#include "source/common/network/utility.h"
+#include "source/extensions/common/tap/tap.h"
+
+#include "contrib/envoy/extensions/tap_sinks/udp_sink/v3alpha/udp_sink.pb.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace TapSinks {
+namespace UDP {
+
+namespace TapCommon = Extensions::Common::Tap;
+
+class UdpTapSink : public TapCommon::Sink {
+public:
+  UdpTapSink(const envoy::extensions::tap_sinks::udp_sink::v3alpha::UdpSink& config);
+  // below one is only for UT because
+  // it can't control return value of writePacket if not add below function.
+  UdpTapSink(Network::UdpPacketWriterPtr&& utUdpPacketWriter)
+      : udp_packet_writer_(std::move(utUdpPacketWriter)) {
+    std::string utUdpServerAddress("127.0.0.1");
+    udp_server_address_ =
+        Network::Utility::parseInternetAddressNoThrow(utUdpServerAddress, 8080, false);
+  }
```

Comment:
If this is only used for unit, it would be better to create a derived class in the unit test files and move this method to the test files.

Or it would be better to let the constructor to take both the address and writer.